internal/driver/glfw/window.go

```diff
@@ -129,7 +129,9 @@ func (w *window) detectTextureScale() float32 {
 }
 
 func (w *window) Show() {
-	go w.doShow()
+	go func() {
+		runOnMain(w.doShow)
+	}()
```

Comment:
Hmm, not to avoid waiting but to avoid blocking. Window creation happens when the main thread kicks off so waiting here would deadlock.
Does the request still stand? we can have a no waiting runOnMain, but I think it's specific to this line...

Reply:
Not waiting on it to finish and not blocking is the same thing though, isn't it? Right now you are spinning up a new goroutine just to have runOnMain wait for the queued function to finish.

Reply:
As I understand it, the runOnMain function queues the function to run on main and then waits for main to report back that it has completed. If we have a version of runOnMain that doesn't wait for main to report back, we would achieve the same non-blocking call as we do here but without the performance penalty (allocating a new stack and scheduling overhead) of launching a new goroutine.

Reply:
It has two code paths... when the app is running it can take the path of queuing the event and not waiting.
However before then the function is called directly. So to avoid that deadlock I have to move the goroutine into that instead. In that case we have more complex code to handle the possibility of a `nil` done but have still not eliminated the goroutine.

Having looked into it to make that summary I'm not sure it's worth it. The overhead is only during a window show which isn't exactly called often.

Reply:
My mental model of the code tells me that this should be rather simply but I'll have a look at the code and see if I can put something simple together. I think we might just not be on the same page about what I mean 😅

Reply:
Hmm. You are right about the blocking vs waiting, I think.

However, I am a bit confused by this code. It seems like the initial show of the window happens before we are running so `runOnMain()` is running the function directly instead of queuing it onto main. This does however seem entirely counter intuitive when you are launching the function within a goroutine because now the function is actually called on the goroutine and not actually on main (unlike what the function name tells you). This does not seem to be intended? It should likely induce races.

Change this line to `go w.doShow()` and you will see `cmd/hello` still works exactly like it is supposed to. fyne_demo does not work but I suppose it is showing a second window after the main thread is started?